src/explore-education-statistics-common/src/modules/charts/components/hooks/useLegend.ts

```diff
@@ -1,20 +1,20 @@
 import omit from 'lodash/omit';
 import { useCallback, useState } from 'react';
-import { LegendProps } from 'recharts';
+import { LegendProps, DefaultLegendContentProps } from 'recharts';
 import { ContentType } from 'recharts/types/component/DefaultLegendContent';
 
 export default function useLegend(): [LegendProps | undefined, ContentType] {
   const [legendProps, setLegendProps] = useState<LegendProps>();
 
   const renderLegend: ContentType = useCallback(
-    nextProps => {
+    (nextProps: DefaultLegendContentProps) => {
       const nextLegendProps = omit(nextProps, 'content');
 
       // Need to do a deep comparison of the props to
       // avoid falling into an infinite rendering loop.
       if (JSON.stringify(legendProps) !== JSON.stringify(nextLegendProps)) {
         setTimeout(() => {
-          setLegendProps(nextLegendProps);
+          setLegendProps(nextLegendProps as LegendProps);
```

Comment:
Again not a blocking comment, just a personal preference:

I try to avoid `as` casts wherever possible, and would personally opt for changing the above lines to: 
`      const nextLegendProps: LegendProps = {
        ...omit(nextProps, 'content'),
        width: nextProps.width ? Number(nextProps.width) : undefined,
        height: nextProps.height ? Number(nextProps.height) : undefined,
      };`
  But I appreciate the probability of height/weight containing non-numeric strings is pretty low :D

Reply:
Yeah that seems like a better solution, I've changed it as you suggest.